File: web/models/conversation-user-statistics.ts
export type UserStatFieldSettingsPayload = {
  show_session_count: boolean
  show_call_count: boolean
  show_unresolved_ticket_count: boolean
  show_total_ticket_count: boolean
}

export type UserStatFieldSettings = UserStatFieldSettingsPayload & {
  id: number | null
  tenant_id: number | null
  configured: boolean
  updated_by_id: number | null
  updated_by_name: string | null
  updated_at: string | null
}

export type ConversationUserStatisticItem = {
  key: 'calls' | 'sessions' | 'tickets'
  value: number | null
  unresolved_value: number | null
  total_value: number | null
}

export type ConversationUserStatistics = {
  conversation_id: number
  user_id: number | null
  items: ConversationUserStatisticItem[]
}

File: server/app/schemas/conversation_user_statistics.py
"""
Schemas for workspace user statistic display settings and counts.
"""
from datetime import datetime
from typing import Literal

from pydantic import BaseModel, ConfigDict


class ConversationUserStatFieldSettingsPayload(BaseModel):
    model_config = ConfigDict(extra="forbid")

    show_session_count: bool = True
    show_call_count: bool = True
    show_unresolved_ticket_count: bool = True
    show_total_ticket_count: bool = True


class ConversationUserStatFieldSettingsResponse(ConversationUserStatFieldSettingsPayload):
    id: int | None = None
    tenant_id: int | None = None
    configured: bool
    updated_by_id: int | None = None
    updated_by_name: str | None = None
    updated_at: datetime | None = None


class ConversationUserStatisticItem(BaseModel):
    key: Literal["calls", "sessions", "tickets"]
    value: int | None = None
    unresolved_value: int | None = None
    total_value: int | None = None


class ConversationUserStatisticsResponse(BaseModel):
    conversation_id: int
    user_id: int | None = None
    items: list[ConversationUserStatisticItem]
